pkg/cmd/workflow/run/run_test.go

```diff
@@ -0,0 +1,567 @@
+				RawFields:   []string{"hey=there"},
+				MagicFields: []string{`name=dana scully`, "file=@cool.txt"},
+			},
+		},
+		{
+			name:  "json on STDIN",
+			cli:   "workflow.yml",
+			stdin: `{"cool":"yeah"}`,
+			wants: RunOptions{
+				JSON:     `{"cool":"yeah"}`,
+				Selector: "workflow.yml",
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			io, stdin, _, _ := iostreams.Test()
+			if tt.stdin == "" {
+				io.SetStdinTTY(tt.tty)
+			} else {
+				stdin.WriteString(tt.stdin)
+			}
+			io.SetStdoutTTY(tt.tty)
+
+			f := &cmdutil.Factory{
+				IOStreams: io,
+			}
+
+			argv, err := shlex.Split(tt.cli)
+			assert.NoError(t, err)
+
+			var gotOpts *RunOptions
+			cmd := NewCmdRun(f, func(opts *RunOptions) error {
+				gotOpts = opts
+				return nil
+			})
+			cmd.SetArgs(argv)
+			cmd.SetIn(&bytes.Buffer{})
+			cmd.SetOut(ioutil.Discard)
+			cmd.SetErr(ioutil.Discard)
+
+			_, err = cmd.ExecuteC()
+			if tt.wantsErr {
+				assert.Error(t, err)
+				if tt.errMsg != "" {
+					assert.Equal(t, tt.errMsg, err.Error())
+				}
+				return
+			}
+
+			assert.NoError(t, err)
+
+			assert.Equal(t, tt.wants.Selector, gotOpts.Selector)
+			assert.Equal(t, tt.wants.Prompt, gotOpts.Prompt)
+			assert.Equal(t, tt.wants.JSON, gotOpts.JSON)
+			assert.Equal(t, tt.wants.Ref, gotOpts.Ref)
+			assert.ElementsMatch(t, tt.wants.RawFields, gotOpts.RawFields)
+			assert.ElementsMatch(t, tt.wants.MagicFields, gotOpts.MagicFields)
+		})
+	}
+}
+
+func Test_magicFieldValue(t *testing.T) {
+	f, err := ioutil.TempFile("", "gh-test")
+	if err != nil {
+		t.Fatal(err)
+	}
+	fmt.Fprint(f, "file contents")
+	f.Close()
+	t.Cleanup(func() { os.Remove(f.Name()) })
+
+	io, _, _, _ := iostreams.Test()
+
+	type args struct {
+		v    string
+		opts RunOptions
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    interface{}
+		wantErr bool
+	}{
+		{
+			name:    "string",
+			args:    args{v: "hello"},
+			want:    "hello",
+			wantErr: false,
+		},
+		{
+			name: "file",
+			args: args{
+				v:    "@" + f.Name(),
+				opts: RunOptions{IO: io},
+			},
+			want:    "file contents",
+			wantErr: false,
+		},
+		{
+			name: "file error",
+			args: args{
+				v:    "@",
+				opts: RunOptions{IO: io},
+			},
+			want:    nil,
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := magicFieldValue(tt.args.v, tt.args.opts)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("magicFieldValue() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if tt.wantErr {
+				return
+			}
+			assert.Equal(t, tt.want, got)
+		})
+	}
+}
+
+func TestRun(t *testing.T) {
+	minimalYAMLContent := []byte(`
+name: minimal workflow
+on: workflow_dispatch
+jobs:
+  yell:
+    runs-on: ubuntu-latest
+    steps:
+      - name: do a yell
+        run: |
+          echo "AUUUGH!"
+`)
+	encodedMinimalYAMLContent := base64.StdEncoding.EncodeToString(minimalYAMLContent)
```

Comment:
Instead of enumerating several flavors of workflow YAMLs in this test, it might be simpler to create a more focused test just for `findInputs` and do it there

Reply:
i'm happy to add some more tests but at least in this instance the two yaml files are doing double duty...the minimal one exercises the `collectInputs` codepath when there are no inputs to collect (in addition to testing out the `on: workflow_dispatch` syntax). It's definitely sloppy and an artifact of my rushing some of this work.